From a3159564cec9a791ffc352f115f08a1cba933c07 Mon Sep 17 00:00:00 2001
From: "Ziqian SUN (Zamir)" <zsun@redhat.com>
Date: Fri, 25 Aug 2017 13:47:49 +0800
Subject: [PATCH] trace-cmd report: Fix hash.key in option -w

BZ: BZ1480770

In add_wakeup function the hash list is created with hash.key = value.
This makes add_sched function will never find a match with the key which
results in trace-cmd report will never show the average latency with the
following command:
trace-cmd report -w -F 'sched_switch,sched_wakeup'
With this patch, the command will show average latency as it used to be
in v2.4.2 like

Average wakeup latency: 28.460 usecs
Maximum Latency: 669.268 usecs at timestamp: 16337.629767
Minimum Latency: 2.153 usecs at timestamp: 16337.533735

RT task timings:

Average wakeup latency: 23.106 usecs
Maximum Latency: 57.482 usecs at timestamp: 16337.187531
Minimum Latency: 4.127 usecs at timestamp: 16336.945209

Link: http://lkml.kernel.org/r/1503640069-23069-1-git-send-email-zsun@redhat.com

Reported-by: Joe Mario <jmario@redhat.com>
Signed-off-by: Ziqian SUN (Zamir) <zsun@redhat.com>
Signed-off-by: Steven Rostedt (VMware) <rostedt@goodmis.org>
---
 trace-read.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/trace-read.c b/trace-read.c
index ebb23681ab17..350a8434ea06 100644
--- a/trace-read.c
+++ b/trace-read.c
@@ -610,7 +610,7 @@ static void add_wakeup(unsigned int val, unsigned long long start)
 	info = malloc(sizeof(*info));
 	if (!info)
 		die("Failed to allocate wakeup info");
-	info->hash.key = val;
+	info->hash.key = key;
 	info->start = start;
 	trace_hash_add(&wakeup_hash, &info->hash);
 }
-- 
2.9.5

